scripts/utils/setup-roles.ts

```diff
@@ -88,105 +88,127 @@ export async function main(
 
     console.log(SECTION_SEPARATOR);
 
-    loanCore = await ethers.getContractAt("LoanCore", LOAN_CORE_ADDRESS);
-    factory = await ethers.getContractAt("VaultFactory", VAULT_FACTORY_ADDRESS);
-    lenderNote = await ethers.getContractAt("PromissoryNote", LENDER_NOTE_ADDRESS);
-    borrowerNote = await ethers.getContractAt("PromissoryNote", BORROWER_NOTE_ADDRESS);
-
-    const gasLimit = 300000;
-
     // grant correct permissions for promissory note
     // giving to user to call PromissoryNote functions directly
     for (const note of [borrowerNote, lenderNote]) {
-        await note.connect(deployer).initialize(loanCore.address, {
-            gasLimit,
-        });
+        await note.initialize(LOAN_CORE_ADDRESS);
     }
 
-    // change loancore's owner from deployer to admin
-    const updateOwner = await loanCore.grantRole(DEFAULT_ADMIN_ROLE, admin.address, {
-        gasLimit,
-    });
+    console.log(`borrowerNote and lenderNote have initalized loanCore at address: ${LOAN_CORE_ADDRESS}`);
+    console.log(SUBSECTION_SEPARATOR);
+
+    // grant LoanCore the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateLoanCoreAdmin = await loanCore.grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
+    await updateLoanCoreAdmin.wait();
 
-    await updateOwner.wait();
     console.log(`loanCore has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
+    console.log(SUBSECTION_SEPARATOR);
 
     // grant LoanCore admin fee claimer permissions
-    const updateLoanCoreFeeClaimer = await loanCore.grantRole(FEE_CLAIMER_ROLE, admin.address, {
-        gasLimit,
-    });
-
+    const updateLoanCoreFeeClaimer = await loanCore.grantRole(FEE_CLAIMER_ROLE, ADMIN_ADDRESS);
     await updateLoanCoreFeeClaimer.wait();
+
     console.log(`loanCore has granted fee claimer role: ${FEE_CLAIMER_ROLE} to address: ${ADMIN_ADDRESS}`);
+    console.log(SUBSECTION_SEPARATOR);
+
+    // grant VaultFactory the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateVaultFactoryAdmin = await factory.grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
+    await updateVaultFactoryAdmin.wait();
+
+    console.log(`vaultFactory has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
+    console.log(SUBSECTION_SEPARATOR);
+
+    // grant originationContoller the owner role to enable authorizeUpgrade onlyOwner
+    const updateOriginationControllerAdmin = await loanCore.grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
```

Comment:
What does `await loanCore.grantRole(ADMIN_ROLE, ADMIN_ADDRESS);` have to do with the `OriginationController`? We are granting the roles on `LoanCore` here.....this doesn't interact with the originationcontroller at all